app/controllers/pieces_controller.rb

```diff
@@ -12,10 +12,16 @@ def update
       flash[:danger] = "#{@piece.color.capitalize} #{@piece.type} move to X#{x}/Y#{y} was invalid"
     end
 
+    if @piece.game.en_passant_opportunity_active?
+      @piece.game.update_attribute(:allows_en_passant_capture?, true)
+    else
+      @piece.game.update_attribute(:allows_en_passant_capture?, false) if @piece.game.allows_en_passant_capture?
+    end
+
```

Comment:
Generally, I've seen conditional logic for updating the db in models. I've understood it was best practice to use the controller to authenticate the request but the model is ultimately the device used to make the change if needed. This was something that was pointed out to me by my mentor although I can't remember the reason why this was considered best practice. So like I said I was just curious about it. I have no critique for your code. Just want to better understand it as it stood out from how I normally see controllers written.